### ISSUE TITLE
Is it not better to lineup the condition with the meta.condition

### ISSUE THREAD

**Issue by henderiw** [GoogleContainerTools/kpt#3657]
### Expected behavior

right now we have to translate from porch.condition to metav1 condition as there are many functions available through the metav1.condition we can reuse. We can also have the last transition time in it to keep track what changed when.

### Actual behavior

right now we translate from one to another so we can avoid this

### Information

func convertConditions(conditions []porchv1alpha1.Condition) *[]metav1.Condition {
	var result []metav1.Condition
	for _, c := range conditions {
		result = append(result, metav1.Condition{
			Type:    c.Type,
			Reason:  c.Reason,
			Status:  metav1.ConditionStatus(c.Status),
			Message: c.Message,
		})
	}
	return &result
}

func unconvertConditions(conditions *[]metav1.Condition) []porchv1alpha1.Condition {
	var prConditions []porchv1alpha1.Condition
	for _, c := range *conditions {
		prConditions = append(prConditions, porchv1alpha1.Condition{
			Type:    c.Type,
			Reason:  c.Reason,
			Status:  porchv1alpha1.ConditionStatus(c.Status),
			Message: c.Message,
		})
	}

	return prConditions
}

### Steps to reproduce the behavior

**Comment by mortent** [comment]
yeah, we can consider using the metav1.Condition type. My main concern about it is that it includes the `LastTransitionType` property as a required value. For a controller this is probably not an issue, but we mirror the conditions into the Kptfile, and filling out the timestamp manually is somewhat awkward.
